### PR TITLE
Implement filtering (by IoU) of predictions that significantly overlap with ground truth objects

### PR DESCRIPTION
**Pull request by almazgimaev**

### CODE REVIEW

File: supervisely/nn/inference/inference.py [hunk 1]
```diff
@@ -3459,6 +3488,80 @@ def _add_workflow_input(self, model_source: str, model_files: dict, model_info:
...
+                    after = len(anns[i].labels)
+                    logger.debug(
+                        f"{[i]}: applied NMS with IoU={iou}. Before: {before}, After: {after}. Time: {timer.get_time():.3f}ms"
+                    )
+            logger.debug(
+                f"Applied NMS with IoU={iou} for {len(anns)} images. Time: {batch_timer.get_time():.3f}ms"
+            )
+        return anns
+
+    def apply_nms(self, gt_ann: Annotation, pred_ann: Annotation, iou_threshold: float):
```
Comment (max-unfinity) [inline supervisely/nn/inference/inference.py:3521]: def exclude_already_annotated_predictions(self, gt_ann: Annotation, pred_ann: Annotation, iou_threshold: float):
    """
    Filter out predictions that significantly overlap with ground truth annotations.
    
    This function compares each prediction with ground truth annotations of the same class
    and removes predictions that have an IoU (Intersection over Union) greater than or equal
    to the specified threshold with any ground truth annotation. This is useful for identifying
    new objects that aren't already annotated in the ground truth.
    
    Parameters:
    -----------
    gt_ann : Annotation
        Annotation object containing ground truth labels
    pred_ann : Annotation
        Annotation object containing prediction labels to be filtered
    iou_threshold : float
        IoU threshold (0.0-1.0). Predictions with IoU >= threshold with any 
        ground truth box of the same class will be removed
    
    Returns:
    --------
    Annotation
        A new annotation object containing only predictions that don't significantly
        overlap with ground truth annotations
    
    Notes:
    ------
    - Predictions with classes not present in ground truth will be kept
    - Requires PyTorch and torchvision for IoU calculations
    """

File: supervisely/nn/inference/inference.py [hunk 2]
```diff
@@ -1347,6 +1347,7 @@ def _inference_batch_ids(self, api: Api, state: dict):
                 source=images_np,
                 settings=settings,
             )
+            anns = self._apply_nms_if_needed(api, anns, settings, dataset_id, ids)
```
Comment (max-unfinity) [inline supervisely/nn/inference/inference.py:1350]: Can we add it in only one place - `_inference_auto` ?

File: supervisely/nn/inference/inference.py [hunk 3]
```diff
@@ -3459,6 +3488,80 @@ def _add_workflow_input(self, model_source: str, model_files: dict, model_info:
...
+
+        try:
+            import torch
+            from torchvision.ops import box_iou
+
+        except ImportError:
+            raise ImportError("Please install PyTorch and torchvision to use this feature.")
+
+        def _to_tensor(geom):
+            return torch.tensor([geom.left, geom.top, geom.right, geom.bottom]).float()
```
Comment (max-unfinity) [inline supervisely/nn/inference/inference.py:3538]: to_bbox
